src/raw_stream.rs

```diff
@@ -59,13 +60,20 @@ pub struct RawDevice {
     supported_switch: Option<AttributeSet<SwitchType>>,
     supported_led: Option<AttributeSet<LedType>>,
     supported_misc: Option<AttributeSet<MiscType>>,
+    auto_repeat: Option<[u32; 2]>,
     // ff: Option<AttributeSet<_>>,
     // ff_stat: Option<FFStatus>,
     // rep: Option<Repeat>,
     supported_snd: Option<AttributeSet<SoundType>>,
     pub(crate) event_buf: Vec<libc::input_event>,
 }
 
+#[derive(Debug)]
+pub struct AutoRepeat {
```

Comment:
If you make this #[repr(C)] it might make the internal storage more clear, as you could store it directly in RawDevice. You would have to do some ugly slice casting in the `eviocgrep` calls, but it might be worth the tradeoff.

Reply:
I agree. I've updated the code accordingly.